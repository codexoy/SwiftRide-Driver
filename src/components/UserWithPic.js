import React from "react"
import { View, Text, Image, StyleSheet } from "react-native"

export default class UserWithPic extends React.Component {
	render() {
		const width = this.props.width || 34
		const img =
			"https://images.unsplash.com/photo-1535643302794-19c3804b874b?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=2134&q=80"
		return (
			<View style={{ ...styles.container, ...this.props.style }}>
				<View style={this.props.imageContainerStyle}>
					<Image
						style={{
							...this.props.imageStyle,
							...{
								borderRadius: width / 2,
								width: width,
								height: this.props.height || 34
							}
						}}
						source={{
							uri: this.props.img || img
						}}
					/>
				</View>
				<View style={styles.column}>
					<Text style={{ ...styles.name, ...this.props.nameStyle }}>
						{this.props.name}
					</Text>
					<View>{this.props.extra}</View>
				</View>
			</View>
		)
	}
}

const styles = StyleSheet.create({
	container: {
		flex: 1,
		flexDirection: "row"
	},
	name: {
		fontWeight: "bold"
	},
	column: {
		flexDirection: "column",
		marginLeft: 16,
		alignItems: "flex-start"
	}
})
