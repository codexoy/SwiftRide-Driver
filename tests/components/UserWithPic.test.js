import React from "react"

import renderer from "react-test-renderer"

import UserWithPic from "../../src/components/UserWithPic"

describe("UserWithPic", () => {
	test("view renders correctly", () => {
		const url = "http://example.com"
		const tree = renderer
			.create(
				<UserWithPic
					style={{ width: 100 }}
					width={100}
					height={100}
					img={url}
					nameStyle={{ maxHeight: 30 }}
					extra="test"
					name="test"
				/>
			)
			.toJSON()
		expect(tree).toMatchSnapshot()
	})
})
