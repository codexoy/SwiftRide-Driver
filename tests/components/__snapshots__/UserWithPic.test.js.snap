// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserWithPic view renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "row",
      "width": 100,
    }
  }
>
  <View>
    <View
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Object {
            "borderRadius": 50,
            "height": 100,
            "width": 100,
          },
        ]
      }
    >
      <FastImageView
        resizeMode="cover"
        source={
          Object {
            "uri": "https://res.cloudinary.com/hsmkon3br/image/upload/c_thumb,g_face,h_80,w_80/v1554188012/678099-profile-filled-256.png",
          }
        }
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      />
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "flexDirection": "column",
        "marginLeft": 16,
      }
    }
  >
    <Text
      style={
        Object {
          "fontWeight": "bold",
          "maxHeight": 30,
        }
      }
    >
      test
    </Text>
    <View>
      <Text>
        test
      </Text>
    </View>
  </View>
</View>
`;
